File: implementation/scipamato/scipamato-core-entity/src/test/java/ch/difty/scipamato/core/entity/keyword/KeywordTest.java
package ch.difty.scipamato.core.entity.keyword;

import static ch.difty.scipamato.core.entity.keyword.Keyword.KeywordFields.NAME;
import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

import ch.difty.scipamato.core.entity.Jsr303ValidatedEntityTest;

class KeywordTest extends Jsr303ValidatedEntityTest<Keyword> {

    KeywordTest() {
        super(Keyword.class);
    }

    @Override
    protected Keyword newValidEntity() {
        return new Keyword(10, "kw", "so");
    }

    @Override
    protected String getToString() {
        return "Keyword(name=kw, searchOverride=so)";
    }

    @Override
    protected String getDisplayValue() {
        return newValidEntity().getName();
    }

    @Test
    void get() {
        Keyword nt = newValidEntity();
        assertThat(nt.getId()).isEqualTo(10);
        assertThat(nt.getName()).isEqualTo("kw");
        assertThat(nt.getSearchOverride()).isEqualTo("so");
    }

    @Test
    void validatingKeyword_withCompleteInformation_succeeds() {
        verifySuccessfulValidation(newValidEntity());
    }

    @Test
    void validatingNewsletter_withNullIssue_fails() {
        Keyword nt = newValidEntity();
        nt.setName(null);
        validateAndAssertFailure(nt, NAME, null, "{javax.validation.constraints.NotNull.message}");
    }

}

File: implementation/scipamato/core/entity/src/test/kotlin/ch/difty/scipamato/core/entity/keyword/KeywordTest.kt
package ch.difty.scipamato.core.entity.keyword

import ch.difty.scipamato.core.entity.Jsr303ValidatedEntityTest
import ch.difty.scipamato.core.entity.keyword.Keyword.KeywordFields.NAME
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class KeywordTest : Jsr303ValidatedEntityTest<Keyword>(Keyword::class.java) {

    override fun newValidEntity(): Keyword {
        return Keyword(10, "kw", "so")
    }

    override val toString: String = "Keyword(name=kw, searchOverride=so)"
    override val displayValue: String = newValidEntity().name

    @Test
    fun get() {
        val nt = newValidEntity()
        assertThat(nt.id).isEqualTo(10)
        assertThat(nt.name).isEqualTo("kw")
        assertThat(nt.searchOverride).isEqualTo("so")
    }

    @Test
    fun validatingKeyword_withCompleteInformation_succeeds() {
        verifySuccessfulValidation(newValidEntity())
    }

    @Test
    fun validatingNewsletter_withNullIssue_fails() {
        val nt = newValidEntity()
        nt.name = null
        validateAndAssertFailure(nt, NAME, null, "{javax.validation.constraints.NotNull.message}")
    }

}